app/src/ingest_bem_pdfs.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import sys
+import uuid
+from dataclasses import dataclass
+from typing import BinaryIO
+
+from smart_open import open as smart_open
+
+from src.adapters import db
+from src.app_config import app_config
+from src.db.models.document import Chunk, Document
+from src.ingestion.pdf_elements import EnrichedText
+from src.ingestion.pdf_postprocess import group_texts
+from src.util import pdf_utils
+from src.util.file_util import get_files
+from src.util.ingest_utils import process_and_ingest_sys_args
+from src.util.pdf_utils import Heading
+
+logger = logging.getLogger(__name__)
+
+# Print INFO messages since this is often run from the terminal
+# during local development
+logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")
+
+HEADER_PATTERN = r"(BEM\s\d*\s+\d+\sof\s\d+\s+\w.*)"
+
+
+def _get_bem_title(file: BinaryIO, file_path: str) -> str:
+    """
+    Get the BEM number from the file path (e.g., 100.pdf) and the
+    document title from the PDF meta data, then put the document
+    title in title case (e.g., INTRODUCTION EXAMPLE -> Introduction Example)
+    and combine: "BEM 100: Introduction Example"
+    """
+    pdf_title = pdf_utils.get_pdf_info(file)["title"]
```

Comment:
This will throw an exception and interrupt ingestion for the directory if a PDF doesn't have a title:

```suggestion
    pdf_info = pdf_utils.get_pdf_info(file)
    if 'title' not in pdf_info:
        ... # handle this case
    pdf_title = pdf_info["title"]
```

I have a suggestion later to make this function return a dataclass, which enables mypy to catch these